### PR TITLE
backend/rv64: Implement more opcodes

### PR DESCRIPTION
**Pull request by ksco**
Adds minimal implementation for test case `arm: Unintended modification in SetCFlag` to work.

### CODE REVIEW

File: externals/mcl/include/mcl/assert.hpp [hunk 1]
```diff
@@ -24,6 +24,7 @@ template<typename... Ts>
 }  // namespace mcl::detail
 
 #define UNREACHABLE() ASSERT_FALSE("Unreachable code!")
+#define UNIMPLEMENTED() ASSERT_FALSE("Unimplemented at {}:{} ", __FILE__, __LINE__);
```
Comment (merryhime) [inline externals/mcl/include/mcl/assert.hpp:27]: This change is occuring in a separate project and should be it's own commit so that it can be pushed upstream.

**Comment by merryhime** [comment]
Don't worry about the mcl change, will update myself when I next update mcl

**Comment by ksco** [comment]
> Don't worry about the mcl change, will update myself when I next update mcl

Thank you, when you do that, could you also rebase feature/rv64 to the latest?

**Comment by merryhime** [comment]
Decided to push to master.
